### PR TITLE
CSS-2279 Initial workflow definitions.

### PR DESCRIPTION
**Pull request by alesstimec**
## Description

Initial workflow definitions to build charms, snap and oci image.

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests
- [ ] Independent change*

### CODE REVIEW

File: .github/workflows/charm.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        description: "Charm to build."
+        default: "candid"
+        required: false
+        type: choice
+        options:
+          - candid
+          - candid-k8s
+
+env:
+  GH_USER: alesstimec
```
Comment (ale8k) [inline .github/workflows/charm.yaml]: Would put this into a repository secret

File: .github/workflows/setupgo118amd64/action.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: 'Setup go and vendor'
+description: 'Setup go with our auth'
```
Comment (ale8k) [inline .github/workflows/setupgo118amd64/action.yaml]: Checkout livepatches updated version of this, ill eventually put it in a reusable one for us so we can share it

**Review by ale8k — APPROVED** [review]
LGTM minor comments, also loving the lint annotations ; )

File: .github/workflows/charm.yaml (hunk 1 above)
Comment (alesstimec) [reply .github/workflows/charm.yaml]: done

File: .github/workflows/setupgo118amd64/action.yaml (hunk 2 above)
Comment (alesstimec) [reply .github/workflows/setupgo118amd64/action.yaml]: done

**Pull request merged**